src/OptInExporter.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\campaignion_logcrm;
+
+use Drupal\campaignion_newsletters\NewsletterList;
+use Drupal\little_helpers\Webform\Submission;
+
+/**
+ * Exporter for opt-in data in a submission.
+ */
+class OptInExporter {
+
+  /**
+   * Mapping of list_id (DB primary key) to global list identifier.
+   *
+   * @var string[]
+   */
+  protected $listIdentifiers;
+
+  /**
+   * The user’s IP-address.
+   *
+   * @var string
+   */
+  protected $ipAddress;
+
+  /**
+   * Map lists to their global identifiers.
+   *
+   * @param \Drupal\campaignion_newsletters\NewsletterList[] $lists
+   *   The lists which’s identifers to get, keyed by list ID.
+   *
+   * @return string[]
+   *   Identifiers keyed by list ID. The identifier consists of a part that
+   *   identifies the provider (eg. mailchimp, cleverreach, …) and the
+   *   provider’s list identifier.
+   */
+  public static function listMap(array $lists = NULL) {
+    $lists = $lists ?? NewsletterList::listAll();
+    return array_map(function ($list) {
+      $source = strtolower(explode('-', $list->source, 2)[0]);
+      if ($source == 'optivo') {
+        $source = 'episerver';
```

Comment:
Hardcoding the identifiers here is a bit smelly, don't you think? OTOH that's the module for such mappings, right?

Reply:
Any plans to add the generic part to e.g. `campaignion_newsletters_provider_info` so there is no need to `explode()` the source here?

(Just curious, not blocking approval)

Reply:
Yes, I had the very same idea. It would need a change in all modules implementing providers (4 x campaignion_newsletters_* and 2 versions of dadiapi) and a way to read this new property. I’ve decided to not do this right now.